src/modules/qt/gps_drawing.cpp

```diff
@@ -0,0 +1,647 @@
+					p.setPen( pen_solid_color0 );
+				else if (color_style == gpsg_color_by_solid_past_future)
+					p.setPen( pen_solid_color1 );
+
+				if (!dots_only)
+					p.drawLine(QPointF(now_pt.x, now_pt.y), QPointF(next_pt.x, next_pt.y));
+			}
+			else
+			{
+				//if invalid point, consider the entire line "future"
+				if (color_style == gpsg_color_by_solid_past)
+					p.setPen( pen_thin_color1 );
+				else if (color_style == gpsg_color_by_solid_future)
+					p.setPen(pen_solid_color0 );
+				else if (color_style == gpsg_color_by_solid_past_future)
+					p.setPen(pen_solid_color1);
+
+				if (dots_only)
+					p.drawPoint(QPointF(crt_pt.x, crt_pt.y));
+				else
+					p.drawLine(QPointF(crt_pt.x, crt_pt.y), QPointF(next_pt.x, next_pt.y));
+			}
+		}
+		else //= full segment lines not intersecting now_dot
+		{
+			//IMPORTANT: the function call without QPointF() cast loses precision due to int!! fun times debugging the small random wiggles from this one
+			if (dots_only)
+				p.drawPoint(QPointF(crt_pt.x, crt_pt.y));
+			else
+				p.drawLine(QPointF(crt_pt.x, crt_pt.y), QPointF(next_pt.x, next_pt.y));
+		}
+	}
+
+
+	//draw the current value in the bot-right corner, big bold white text
+	if (mlt_properties_get_int(properties, "show_now_text"))
+	{
+		double now_val = get_crtval_bysrc(filter, 0, 0, &gps_now);
+		if (now_val != GPS_UNINIT)
+		{
+			QRectF now_text_rect = QRectF( rect.x, rect.y + rect.h/2, rect.w, rect.h/2 );
+			QFont font = p.font();
+			font.setPixelSize( rect.w/15 );
+			font.setWeight( QFont::Bold );
+			p.setFont(font);
+			p.setPen( Qt::white );
+
+			now_val = convert_bysrc_to_format(filter, now_val);
+			QString now_text = QString::number(now_val, 'f', decimals_needed_bysrc(filter, now_val));
+			if (pdata->graph_data_source == gspg_location_src) 
+			{
+				now_val = get_crtval_bysrc(filter, 0, gpsg_longitude_id, &gps_now);
+				now_val = swap_180_if_needed(now_val);
+				now_text += ", " + QString::number(now_val, 'f', decimals_needed_bysrc(filter, now_val));
+			}
+			now_text += QString(legend_unit);
+			
+			p.drawText(now_text_rect, Qt::AlignBottom | Qt::AlignRight, now_text);
+		}
+	}
+
+	//restore to before legend + main_graph
+	p.restore();
+	//set color so by default the now_dot will be the same color as nearby graph
+	p.setPen(last_graph_pen);
+
+	//draw a circle for current point
+	if (mlt_properties_get_int(properties, "show_now_dot") && pdata->graph_type <= 1)
+		draw_now_dot(filter, frame, p, used_crops);
+}
+
+//draws a speedometer from the current data source (for map it displays % of total)
+void draw_main_speedometer(mlt_filter filter, mlt_frame frame, QPainter &p, s_base_crops &used_crops)
+{
+	private_data* pdata = (private_data*)filter->child;
+	mlt_rect rect = pdata->img_rect;
+	mlt_properties properties = MLT_FILTER_PROPERTIES( filter );
+	int color_style = mlt_properties_get_int( properties, "color_style" ) ;
+	int thickness = qAbs( mlt_properties_get_int( properties, "thickness" ) );
```

Comment:
`color_style` and `thickness` are never used.

Reply:
I've only commented them out for now, I'm considering using them in a future update.